src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -70,6 +70,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="GenerateBuildDependencyFile"
           DependsOnTargets="_DefaultMicrosoftNETPlatformLibrary"
+          BeforeTargets="_CheckForCompileOutputs"
```

Comment:
The `IncrementalClean` target runs during a normal build (`CoreBuild` depends on it).  `IncrementalClean` depends on `_CleanGetCurrentAndPriorFileWrites`, which depends on `_CheckForCompileOutputs`.  So it's confusing (which is why I asked for Rainier's review), but I think it's doing the right thing.

> Did you prefer this for a reason?

I think it was because `GenerateBuildDependencyFile` also writes files to the output folder, so I thought it might make sense to be before the `CopyFilesToOutputDirectory` target, which depends on `_CheckForCompileOutputs`.